Solar_radiation_scripts/retrieve_DEOS_data_v2.R

```diff
@@ -2,72 +2,102 @@ library(XML)
-
-  # export message so you know what's going on
-  message('Retrieving data for ', all_days[i])
-
-  # get that day's data
-  dat_list <- XML::readHTMLTable(new_url, stringsAsFactors = FALSE)
-  dat <- dat_list[[2]][c(1,3,6,17,19)] %>%
-    mutate(date = all_days[i])
-
-  dat_all <- rbind(dat, dat_all)
+# check wd
+getwd()
+
+# Set wd
+setwd('..')
+
+##### Retrieve DEOS data from all DEOS stations within DRB #####
+# Load .csv of DEOS stations within DRB
+DEOS_stations <- read.csv('data-raw/DEOS/DEOS_DRB.csv', header = TRUE)
+
+# Define function to retrieve data
+retrieve_DEOS <- function(input_dataframe, start_yr, end_yr){
+  # get a sequence of dates
+  all_days <- seq(as.Date(paste0(start_yr,"-01-01")), as.Date(paste0(end_yr,"-12-31")), by="days")
+  # create empty data frame for data from all stations
+  raw_DEOS <- data.frame()
+  # set base url
+  base_url <- 'http://deos.udel.edu/odd-divas/station_daily.php?network=DEOS&'
+  # iterate through rows to....
+  for (row in 1:nrow(input_dataframe)){
+    # create empty data frame for data from each station
+    raW_DEOS_station <- data.frame()
+    # Set station extension
+    this_station <- input_dataframe[row, "Station"]
+    station_extension <- sprintf('station=%s&', this_station)
+    # set latitude and longitude
+    this_latitude <- input_dataframe[row, "Lat_dd"]
+    this_longitude <- input_dataframe[row, "Lon_dd"]
+    # Then, iterate through all days in year to...
+    for (i in 1:length(all_days)) {
+      # build final url
+      date_extension <- sprintf('month=%s&day=%s&year=%s', lubridate::month(all_days[i]), lubridate::day(all_days[i]), lubridate::year(all_days[i]))
+      final_url <- paste0(base_url, station_extension, date_extension)
+      # export message so you know what's going on
+      message('Retrieving data for station ', this_station, ": ", all_days[i])
+      # get that day's data
+      dat_list <- XML::readHTMLTable(final_url, stringsAsFactors = FALSE)
+      dat <- dat_list[[2]][c(1,3,6,17,19)] %>%
+        mutate(Date = all_days[i]) %>%
+        mutate(Station = this_station, Latitude = this_latitude, Longitude = this_longitude)
+      raw_DEOS <- rbind(dat, raw_DEOS)
+      raW_DEOS_station <- rbind(dat, raW_DEOS_station)
+    }
+    # export data for each station
+    message('Exporting all data retrieved for station ', this_station)
+    if (start_yr == end_yr){
+      station_output_filename <- paste0("data-raw/DEOS/", this_station, "_", start_yr, "_daily.csv")
+    } else {
+      station_output_filename <- paste0("data-raw/DEOS/", this_station, "_", start_yr, "-", end_yr, "_daily.csv")
+    }
+    write.csv(raW_DEOS_station, station_output_filename, row.names = FALSE)
+  }
+  # Export all retrieved data for all stations
+  message('Exporting all data retrieved for all stations')
+  names(raw_DEOS) <- c('Hour', 'Temp_degC', 'Relative_humidity_%', 'Solar_radiation_wattsperm2', 'Rainfall_mm', 'Date', 'Station', 'Latitude', 'Longitude')
```

Comment:
maybe don't include `%` sign as it might be a special character?

Reply:
Good point. I'll make that change.